### PR TITLE
[Feature] Changed to use tsdx, tslint changed to eslint and others

### PR DESCRIPTION
**Pull request by ketsappi**
## Description

- Changed the build system to use [tsdx](https://github.com/jaredpalmer/tsdx)
  - tsdx is built on top of rollup
- `styled-components` is now a peer dependency and needs to be installed by the user of this library
  - Also telling about this and the requirement of the `react` in the README's **Install**-section
- tslint changed to eslint
  - some changes done based on new rules / adjustments
- Dropped the usage instructions of bare accessible components mentioning from the README as it is not correctly working as intended
  - this will be added back later on, when it is working again 
- updated some dependencies with newer version
- changed some dependencies to new version as old one was deprecated

## Related Issue

Relates to following issues: #226, #202 & #150 

## Motivation and Context

- Making it simpler to build the library
  - tsdx will create all the needed builds
- Getting rid of old tslint
- Steps taken to support SSR; for what we are able to do on the library's side

## How Has This Been Tested?

- Locally in various SSR-template projects
- `yarn validate`

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,9 +1,76 @@
...
+    "@typescript-eslint/explicit-function-return-type": "off",
+    "@typescript-eslint/no-empty-interface": [
+      "warn",
+      {
+        "allowSingleExtends": true
+      }
+    ],
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/no-inferrable-types": [
+      "warn",
```
Comment (aappoalander) [inline .eslintrc]: Not sure if it's a good idea to use warnings since those might not work in Circle CI. At least worth checking before merging this setup.

File: package.json [hunk 2]
```diff
@@ -1,11 +1,10 @@
 {
   "name": "suomifi-ui-components",
-  "version": "0.4.1",
+  "version": "0.5.0-alpha.1",
```
Comment (aappoalander) [inline package.json]: Is this a good idea to change the version with this PR? I'd propose to keep it as is and make a separate PR and a tag for 0.5.0.

File: src/components/Form/ToggleButton.tsx [hunk 3]
```diff
@@ -55,7 +55,7 @@ export class ToggleButton extends Component<ToggleProps> {
 
     if (!!toggleInputProps || !!toggleInputComponent) {
       logger.error(
-        `ToggleButton does not utilize 'toggleInputProps' and 'toggleInputComponent' props.`,
+        'ToggleButton does not utilize toggleInputProps and toggleInputComponent props.',
```
Comment (aappoalander) [inline src/components/Form/ToggleButton.tsx]: Escaping characters should be possible here? \'

File: src/core/theme/index.ts [hunk 4]
```diff
@@ -58,13 +62,13 @@ const internalTokens = {
...
-  colors,
-  spacing,
+  colors: colorTokens,
+  spacing: spacingTokens,
   typography: suomifiDesignTokens.typography,
   values: suomifiDesignTokens.values,
 };
 
-export const defaultTokens = {
+export const defaultThemeTokens = {
```
Comment (aappoalander) [inline src/core/theme/index.ts]: Is this exported outside the library / part of API? If it is, then I'd try to keep it as it was.

File: package.json (hunk 2 above)
Comment (ketsappi) [reply package.json]: Good point. Didn't notice the version number increase that we made for the alpha release when making PR.

Let's do like you suggested; keeping it at the previous `0.4.1` in this PR and increase it in the PR of `release/v0.5.0` little later.

File: src/components/Form/ToggleButton.tsx (hunk 3 above)
Comment (ketsappi) [reply src/components/Form/ToggleButton.tsx]: I changed this when looking through the warning given by eslint and thought that there might be no reason to use single quotes around the props to give anything more to the reader of warning message.

This current eslint rule allows also the previous version. 
https://github.com/vrk-kpa/suomifi-ui-components/blob/dca31e6cc60c34523004d1c599ee5102cfba084d/.eslintrc#L100

Maybe we can return this highlighted code to its previous value if it is giving more to the reader?

File: src/core/theme/index.ts (hunk 4 above)
Comment (ketsappi) [reply src/core/theme/index.ts]: I don't see that it is used outside the library. Here is the two places where it is used:

---

1.) 

https://github.com/vrk-kpa/suomifi-ui-components/blob/dca31e6cc60c34523004d1c599ee5102cfba084d/src/core/theme/utils/defaults.ts#L11-L12

---

2.) 

https://github.com/vrk-kpa/suomifi-ui-components/blob/dca31e6cc60c34523004d1c599ee5102cfba084d/src/core/theme/tokens.test.tsx#L8

File: .eslintrc (hunk 1 above)
Comment (ketsappi) [reply .eslintrc]: So, should we use errors instead of warnings?
In this particular rule or in all rules? 

Ideally there is no warnings at all, so basically we could use just errors 🤔

Comment (aappoalander) [reply .eslintrc]: Does the CI pipeline pass with these settings?

Comment (aappoalander) [reply .eslintrc]: Seems to work so this is ok for now in my opinion!

File: src/components/Form/ToggleButton.tsx (hunk 3 above)
Comment (ketsappi) [reply src/components/Form/ToggleButton.tsx]: Reverted to the previous value in latest commit.
